### PR TITLE
Resilient prrte master

### PR DESCRIPTION
**Pull request by dong0321**
This is the code used for the paper, fixed all those comments/conflicts/known-bugs in last PR.

Please help review, targeting PRRTE upstream. (Rebased upon PRRTE/master)

### CODE REVIEW

File: ORTE_relilient_Build_Test_Recipe [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- 'orte_abort_non_zero_exit <true|false> (default: false)` controls the job
+  termination after a error happpens.
+
+- 'errmgr_detector_enable <true|false> (default: true)` enable or disable error
+  detection and propagation.
+
+## Tesing
+=====================
+Step 1: salloc -k -n x -w host1,host2...
+        -k, --no-kill do not kill job on node failure
```
Comment (bosilca) [inline ORTE_relilient_Build_Test_Recipe]: During execution can we identify if the --no-kill switch has been provided to the RM ?

File: orte/mca/errmgr/dvm/errmgr_dvm.c [hunk 2]
```diff
@@ -639,7 +774,7 @@ static void proc_errors(int fd, short args, void *cbdata)
          * hosed - so just exit out
          */
         if (ORTE_PROC_MY_NAME->jobid == proc->jobid) {
-            ORTE_ACTIVATE_JOB_STATE(NULL, ORTE_JOB_STATE_DAEMONS_TERMINATED);
+            //ORTE_ACTIVATE_JOB_STATE(NULL, ORTE_JOB_STATE_DAEMONS_TERMINATED);
```
Comment (bosilca) [inline orte/mca/errmgr/dvm/errmgr_dvm.c]: Add a nice comment here about what is happening and why we are not doing the default anymore.

File: orte/mca/ess/base/ess_base_std_orted.c [hunk 3]
```diff
@@ -566,13 +605,27 @@ int orte_ess_base_orted_finalize(void)
...
 
+    if ( NULL != orte_propagate.finalize ) {
+        orte_propagate.finalize();
+    }
+    (void) mca_base_framework_close(&orte_propagate_base_framework);
+
+    if ( NULL != orte_errmgr.finalize ) {
+        orte_errmgr.finalize();
+    }
+    //(void) mca_base_framework_close(&orte_errmgr_base_framework);
```
Comment (bosilca) [inline orte/mca/ess/base/ess_base_std_orted.c]: the errmgr framework never gets closed (only finalized) ?

**Comment by abouteiller** [comment]
This has been merged with upstream. Discussion is completed.

**Pull request closed without merging**